src/functions.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+
+function generateArtistHtml(array $artists): string {
+    if (array_key_exists('name', $artists) 
+    && array_key_exists('yearsLived', $artists) 
+    && array_key_exists('image', $artists) 
+    && array_key_exists('favouriteMedium', $artists)
+    && array_key_exists('knownFor', $artists)
+    && array_key_exists('placeOfBirth', $artists)) {
+        return '<h2>' . $artists['name'] . '</h2>' . 
+        '<p class="years">' . $artists['yearsLived'] . '</p>' .
+        '<img alt="an image of artwork by ' . $artists['name'] . '" class="image" src=' . $artists['image'] . '>' .
+        '<p>Favourite Medium: ' . $artists['favouriteMedium'] . '</p>' . 
+        '<p>Known For: ' . $artists['knownFor'] . '</p>' . 
+        '<p>Place Of Birth: ' . $artists['placeOfBirth'] . '</p>';
+    } else {
+        throw new Exception('Wrong array keys entered'); 
```

Comment:
I don't understand your exception message. Do you mean something like "Missing information"?

Reply:
I have changed it to be more specific, although as this message is for ourselves to diagnose problem I think it should still mention array keys etc?